### ISSUE TITLE
Add option to disable "open this link in browser" popup.

### ISSUE THREAD

**Issue by altsst** [oxen-io/session-desktop#2662]
**Describe the bug**
The session client on android/mac/windows is asking if I want to open a link in the browser.

There should be an **option to disable that popup** and open it directly in the browser.

**To Reproduce**
1. Send a link to someone
2. Click on link to try and open it in the browser

**Screenshots or Logs**
![image](https://user-images.githubusercontent.com/116441438/216809875-eb3524ad-ecb6-4ae8-ba16-1ae0fe45e681.png)

**Other information (please complete the following information):**

* Device: MacOS, Windows, Android
* OS: MacOS 12.6, Windows 10, Android 13
* Session Version or Git commit hash: Session 1.10.4, 991ddd1cbaa1948c865b5e8ad9b2b8137f04641c

**Comment by KeeJef** [comment]
Yeah i think this is a good idea, would support putting an option in the modal, something like "Don't show me this again"

**Comment by VityaSchel** [comment]
> Yeah i think this is a good idea, would support putting an option in the modal, something like "Don't show me this again"

I think it would be better to add "Trust this domain" button which adds this exact hostname to list of trusted and put "Disable link warnings" in settings instead. Users presented with "don't show me this again" will usually click it, because it removes extra time-wasting step for them (risking security), and by adding option to disable these popups to settings we add extra customization for users who took their time to search for settings for this exact option and who take responsibility of clicking on unknown links.

**Comment by VityaSchel** [comment]
I noticed that in current implementation pressing "esc" while in this modal copies link to clipboard. Is this intentional? Looks like bug because SessionConfirm modal is used.

I created separate component `SessionOpenExternalLinkDialog` and separate modal state `OpenExternalLinkModalState`, moved showLinkVisitWarningDialog from SessionConfirm to OpenExternalLinkDialog so we don't mix these two modals in same file and took some design ideas from discord to put all four possible actions in popup: 1) close popup 2) copy link 3) just open 4) open and trust in future

Here is reference from Discord:
<img width="350" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/1e5cc823-1282-4956-9cba-52ceecfebc9d">

Here is what I made:
<img width="350" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/0c2a22f6-cbba-44fd-9822-31d46ea58256">
<img width="350" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/2f762cfb-23ff-4d21-a1ad-bef1cdaa104f">

Here is patch for this:
[0001-chore-refactor-redesign-external-links-modal.patch](https://github.com/oxen-io/session-desktop/files/15138526/0001-chore-refactor-redesign-external-links-modal.patch)

And for trusted websites feature:
<img width="350" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/1f5b69e8-0b6a-4d5e-927e-428e2d3b72a8">
<img width="350" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/cc0c8e76-037c-4879-b0a9-428ed5d79b39">

I added TrustWebsitesController which is a simplier copy of BlockedNumberController that stores trusted hostnames in db by `trusted-websites` key. 

<img width="500" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/9db73022-4f89-4a07-94a3-c49ad6328677">
<img width="500" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/93451a82-5aab-4239-8612-d2359da94b60">
<img width="500" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/c65fe83a-09cd-42ca-a60b-25ffd105d3c9">


By the way, I fixed this issue when options were shrinking because of too small resolution. This is fixed by applying `flex-shrink: 0`. 
<img width="350" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/9cf73ba9-92c5-4650-9041-4797d8c08ce4">
became:
<img width="350" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/c79e9729-eb64-412e-ad93-0f9db0c05f58">
Also this arrow should be reversed (pointing downwards when closed) so I fixed it too
<img width="92" alt="image" src="https://github.com/oxen-io/session-desktop/assets/59040542/446c7c02-6a76-487b-a5cd-26a59d7794a0">

However, I noticed that `trusted-websites` key does not get saved in database, while `blocked` does. I need these changes to be reviewed and approved anyway, so please keep in mind this issue with persisting trusted hostnames string set.

Here is final patch with `yarn ready` executed:

[0001-feat-trusted-websites-list-and-trust-hostname-btn.patch](https://github.com/oxen-io/session-desktop/files/15138694/0001-feat-trusted-websites-list-and-trust-hostname-btn.patch)

And I'll create a pull request when GitHub support detach my bot framework repo and I'll be able to fork session desktop client repo again.

**Comment by KeeJef** [comment]
After looking at this further, i think we want to take this in house as we want to support this feature cross platform and potentially sync the URL list cross platform, which extends this from being a simple PR into something more complex. Tracking this via Jira internally as https://optf.atlassian.net/browse/SES-1808

**Comment by VityaSchel** [comment]
> Thanks for taking a stab at this @VityaSchel , after looking at this further, i think we want to take this in house as we want to support this feature cross platform and potentially sync the URL list cross platform, which extends this from being a simple PR into something more complex. Tracking this via Jira internally as https://optf.atlassian.net/browse/SES-1808

I was finally able to create PR with my code #3088 so feel free to take anything from it, if it helps :)
